MCB-Nucleo/Core/Src/freertos.c

```diff
@@ -0,0 +1,454 @@
+
+  /* creation of motorStateMachi */
+  motorStateMachiHandle = osThreadNew(startMotorStateMachine, NULL, &motorStateMachi_attributes);
+
+  /* creation of getCANVelocity */
+  getCANVelocityHandle = osThreadNew(startGetCANVelocity, NULL, &getCANVelocity_attributes);
+
+  /* creation of getCANBatterySO */
+  getCANBatterySOHandle = osThreadNew(StartSetCANBatterySO, NULL, &getCANBatterySO_attributes);
+
+  /* creation of updateState */
+  updateStateHandle = osThreadNew(StartUpdateState, NULL, &updateState_attributes);
+
+  /* USER CODE BEGIN RTOS_THREADS */
+  /* add threads, ... */
+  /* USER CODE END RTOS_THREADS */
+
+  /* USER CODE BEGIN RTOS_EVENTS */
+  /* add events, ... */
+  /* USER CODE END RTOS_EVENTS */
+
+}
+
+/* USER CODE BEGIN Header_StartDefaultTask */
+/**
+  * @brief  Function implementing the defaultTask thread.
+  * @param  argument: Not used
+  * @retval None
+  */
+/* USER CODE END Header_StartDefaultTask */
+void StartDefaultTask(void *argument)
+{
+  /* USER CODE BEGIN StartDefaultTask */
+  /* Infinite loop */
+  for(;;)
+  {
+	 HAL_GPIO_TogglePin(GPIOA, GPIO_PIN_5);
+     osDelay(500);
+  }
+  /* USER CODE END StartDefaultTask */
+}
+
+/* USER CODE BEGIN Header_getADCValues */
+/**
+* @brief Function implementing the GetADCValues thread.
+* @param argument: Not used
+* @retval None
+*/
+/* USER CODE END Header_getADCValues */
+void getADCValues(void *argument)
+{
+  /* USER CODE BEGIN getADCValues */
+  /* Infinite loop */
+  for(;;)
+  {
+	// Get ADC value for throttle and sets event flag
+	ADC_throttle_val = ReadADC(&hadc1);
+	event_flags.throttle_pressed = ADC_throttle_val > ADC_DEADZONE;
+
+	// Gets ADC value for regen and sets event flags
+	ADC_regen_val = ReadADC(&hadc2);
+	event_flags.regen_pressed = ADC_regen_val > ADC_DEADZONE;
+	if(ADC_regen_val > ADC_DEADZONE)
+		event_flags.cruise_enabled = FALSE;
+
+    osDelay(ADC_POLL_DELAY);
+  }
+  /* USER CODE END getADCValues */
+}
+
+/* USER CODE BEGIN Header_startUpdateFlags */
+/**
+* @brief Function implementing the updateFlags thread.
+* @param argument: Not used
+* @retval None
+*/
+/* USER CODE END Header_startUpdateFlags */
+void startUpdateFlags(void *argument)
+{
+  /* USER CODE BEGIN startUpdateFlags */
+  /* Infinite loop */
+  for(;;)
+  {
+	  event_flags.mech_brake_pressed = HAL_GPIO_ReadPin(MECH_BRAKE_GPIO_Port, MECH_BRAKE_Pin);
+
+	  event_flags.park_enabled = HAL_GPIO_ReadPin(SWITCH_PARK_GPIO_Port, SWITCH_PARK_Pin);
+
+	  event_flags.reverse_enabled = HAL_GPIO_ReadPin(SWITCH_REVERSE_GPIO_Port, SWITCH_REVERSE_Pin);
+
+	  event_flags.velocity_under_threshold = (velocity_of_car < MIN_REVERSE_VELOCITY);
+
+	  event_flags.charge_under_threshold = battery_soc < BATTERY_SOC_THRESHHOLD;
+
+	  osDelay(EVENT_FLAG_UPDATE_DELAY);
+  }
+  /* USER CODE END startUpdateFlags */
+}
+
+/* USER CODE BEGIN Header_startMotorStateMachine */
+/**
+* @brief Function implementing the motorStateMachi thread.
```

Comment:
Make sure to add more details to the documentation. This doesn't really say anything.

For example: "A thread that outputs motor drive commands for all states of the board"

Reply:
Added basic descriptions for each task